cnf-tests/testsuites/pkg/features/features.go

```diff
@@ -0,0 +1,222 @@
+	numaserialconf "github.com/openshift-kni/numaresources-operator/test/e2e/serial/config"
+	perfUtils "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils"
+	perfClean "github.com/openshift/cluster-node-tuning-operator/test/e2e/performanceprofile/functests/utils/clean"
+	ptpClean "github.com/openshift/ptp-operator/test/utils/clean"
+)
+
+var (
+	paoNS            = perfUtils.NamespaceTesting
+	dpdkNS           = namespaces.DpdkTest
+	sroNS            = namespaces.SroTestNamespace
+	gatekeeperNS     = testutils.GatekeeperTestingNamespace
+	sctpNS           = sctp.TestNamespace
+	vrfNS            = vrf.TestNamespace
+	sriovNS          = sriovNamespaces.Test
+	xtu32NS          = namespaces.XTU32Test
+	ovsqosNS         = namespaces.OVSQOSTest
+	tuningcniNS      = security.TestNamespace
+	tuningcniSriovNS = security.SriovTestNamespace
+	bondcniNS        = namespaces.BondTestNamespace
+)
+
+type SuiteFixture interface {
+	Setup() error
+	Cleanup() error
+}
+
+type PAOFixture struct {
+}
+
+func (p *PAOFixture) Setup() error {
+	return namespaces.Create(paoNS, testclient.Client)
+}
+
+func (p *PAOFixture) Cleanup() error {
+	if !discovery.Enabled() {
+		perfClean.All()
+	}
+
+	return namespaces.Delete(paoNS, testclient.Client)
+}
+
+type DPDKFixture struct {
+}
+
+func (p *DPDKFixture) Setup() error {
+	return namespaces.Create(dpdkNS, testclient.Client)
+}
+
+func (p *DPDKFixture) Cleanup() error {
+	return namespaces.Delete(dpdkNS, testclient.Client)
+}
+
+type GatekeeperFixture struct {
+}
+
+func (p *GatekeeperFixture) Setup() error {
+	return namespaces.Create(gatekeeperNS, testclient.Client)
+}
+
+func (p *GatekeeperFixture) Cleanup() error {
+	return namespaces.Delete(gatekeeperNS, testclient.Client)
+}
+
+type SROFixture struct {
+}
+
+func (p *SROFixture) Setup() error {
+	return namespaces.Create(sroNS, testclient.Client)
+}
+
+func (p *SROFixture) Cleanup() error {
+	sro.Clean()
+	return namespaces.Delete(sroNS, testclient.Client)
+}
+
+type PTPFixture struct {
+}
+
+func (p *PTPFixture) Setup() error {
+	return nil
+}
+
+func (p *PTPFixture) Cleanup() error {
+	ptp.Clean()
+	return ptpClean.All()
+}
+
+type SriovFixture struct {
+}
+
+func (p *SriovFixture) Setup() error {
+	return nil
+}
+
+func (p *SriovFixture) Cleanup() error {
+	sriovClean.All()
+
+	err := namespaces.Delete(tuningcniSriovNS, testclient.Client)
+	if err != nil {
+		return err
+	}
+
+	return namespaces.Delete(sriovNS, testclient.Client)
+}
+
+type SCTPFixture struct {
+}
+
+func (p *SCTPFixture) Setup() error {
+	return nil
+}
+
+func (p *SCTPFixture) Cleanup() error {
+	err := namespaces.Clean("default", "testsctp-", testclient.Client)
+	if err != nil {
+		return fmt.Errorf("failed to clean 'default' namespace 'testsctp' prefix")
+	}
+
+	return namespaces.Delete(sctpNS, testclient.Client)
+}
+
+type XTU32Fixture struct {
+}
+
+func (p *XTU32Fixture) Setup() error {
+	return nil
```

Comment:
why we don't create the namespace here?

Reply:
we weren't creating it before. i didn't want to change the existing setup https://github.com/openshift-kni/cnf-features-deploy/blob/57a05afbe836e9210ad638f6abe65f72b569148a/cnf-tests/testsuites/e2esuite/test_suite_test.go#L103

Reply:
ok we can do it in a following PR I think

Reply:
do we need to create it? i guess it is currently done somewhere in the test suite